src/app/components/store-products/store-products.component.ts

```diff
@@ -8,4 +16,29 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class StoreProductsComponent {
+  readonly currentStore$: Observable<StoreModel> = this._activatedRoute.params.pipe(
+    switchMap((data) => {
+      return this._storesService.getOneStore(+data['storeId'])
+    })
+  );
+
+  readonly searchForm: FormGroup = new FormGroup({ search: new FormControl('') });
+
+  readonly searchVal$: Observable<string> = this.searchForm.controls['search'].valueChanges.pipe(
+    startWith(''), shareReplay(1)
+  );
+  readonly currentStoreProducts$: Observable<ProductModel[]> = combineLatest([
+    this.searchVal$,
+    this._productsService.getAllProducts(),
+    this.currentStore$.pipe(shareReplay(1))
```

Comment:
You should to add shareReplay when you declare the variable, not where you use it.